### ISSUE TITLE
Shared System.Net.WebSockets.ClientWebSocket issue

### ISSUE THREAD

**Issue by okarpov** [dotnet/core#7744]
### Description
Shared System.Net.WebSockets.ClientWebSocket reads another task response

I have a instance of System.Net.WebSockets.ClientWebSocket which is used by many Tasks.
At some point i see that System.Net.WebSockets.ClientWebSocket reads data that was suppoused to be read by another task.

Every task just does the following on the shared websocket object:
```
wLoCs.SendAsync(System.Text.Encoding.UTF8.GetBytes($"gd-sfr:{DeviceId}"), System.Net.WebSockets.WebSocketMessageType.Text, false, System.Threading.CancellationToken.None).Wait();

                                var receiveResult = wLoCs.ReceiveAsync(
                                    new ArraySegment<byte>(bufferLoC), System.Threading.CancellationToken.None);
                                Task.WaitAll(receiveResult);
                                string data = System.Text.Encoding.UTF8.GetString(bufferLoC, 0, receiveResult.Result.Count);
```
the byte array bufferLoC is local for each Task

### Configuration
.net core 6
x64

**Comment by wfurt** [comment]
How do you de-multiplex? I don't think there is any guarantee that the receive would be identical to the send. 
cc: @CarnaViire

**Comment by okarpov** [comment]
ok, any example to use it in multitask scenario? (not multithread)
because of use of lock(wLoCs) blocks all 500 Tasks and everything just hangs.
non-use of lock(wLoCs) force to create 500 wLoCs instances or reads another class/task instance data

**Comment by CarnaViire** [comment]
Only one send and one receive is supported on each ClientWebSocket object in parallel. This is stated in the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.net.websockets.clientwebsocket.receiveasync?view=net-6.0).
Calling ReceiveAsync from multiple tasks without any synchronization is not supported and may result in an undefined behavior/exceptions/crashes etc.

It is up to the developer to serialize the access via whatever mechanism works best for them. That could be using a lock/semaphore, it could be guaranteeing the previous operation is awaited/completed before issuing the next, etc.

P.S.: there is an issue to improve the docs to explain better the results of erroneous usage https://github.com/dotnet/dotnet-api-docs/issues/8040

**Comment by wfurt** [comment]
thanks for explanation @CarnaViire
